fastavro/schema.py

```diff
@@ -67,7 +67,7 @@ def extract_named_schemas_into_repo(schema, repo, transformer, parent_ns=None):
 
     namespace, name = schema_name(schema, parent_ns)
 
-    if name and (name not in repo):
```

Comment:
This was added in the initial write support and probably for a reason :)
Is this how the spec says it should be?